modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/action/SaveTableSectionsMVCResourceCommand.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.data.set.views.web.internal.portlet.action;
+
+import com.liferay.frontend.data.set.views.web.internal.constants.FDSViewsPortletKeys;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectEntryService;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseTransactionalMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.io.Serializable;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marko Cikos
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + FDSViewsPortletKeys.FDS_VIEWS,
+		"mvc.command.name=/frontend_data_set_views/save_table_sections"
```

Comment:
I was about to ask why the old `SaveFDSFieldsMVCResourceCommand.java` was not being git-moved to this new action, to preserve history. 

I guess reason is that both will coexist for some time, so a new one was created.

Perhaps we can git move the old to this, make necessary changes (`mcv.command.name`, object definition name, etc), then commit a new file with an exact copy of the older one. This way changes would be much easier to track and this file would keep old history.

If you see this too complicated, we can live without it

Reply:
The problem is that `@Component` property cannot be dynamic. I found that out when I tried to fork init jsp, something like 
```java
"javax.portlet.init-param.view-template=/" + FF ? "data_sets.jsp" : "fds_entries.jsp",
```
Compiler did not like that. That's why I ended up [forking in the jsp](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/data_sets.jsp).

Reply:
What I suggest does not need dynamic `@Component` but rather:

1. Rename `SaveFDSFieldsMVCResourceCommand.java` to `SaveTableSectionsMVCResourceCommand.java`
2. Change the code needed in `SaveTableSectionsMVCResourceCommand.java`  (this includes updating the value of `@Component`) and the rest of things, as needed
3. Create a (new to git) `SaveFDSFieldsMVCResourceCommand.java` from scracth with the exact same code.

This way we'd end up duplicating the ResourceCommands, but the new one would contain all the git history because it was moved from an existing file

Reply:
Ooh there is a difference? :open_mouth: I always assumed git just compares final and previous state, and decides what happened with it's internal logic. I'll try it!

Reply:
I tried this, but it doesn't work. After I amend to commit the newly added `SaveFDSFieldsMVCResourceCommand`, git just resolves content of commit to new `SaveTableSectionsMVCResourceCommand`.